app/src/main/java/com/lftechnology/unito/activity/MainActivity.java

```diff
@@ -0,0 +1,312 @@
+import com.lftechnology.unito.constant.AppConstant;
+import com.lftechnology.unito.fragment.MainFragment;
+import com.lftechnology.unito.helper.OnStartDragListener;
+import com.lftechnology.unito.helper.SimpleItemTouchHelperCallback;
+import com.lftechnology.unito.utils.GeneralUtils;
+import com.lftechnology.unito.utils.SoftKeyBoard;
+import com.squareup.otto.Subscribe;
+
+import java.util.Arrays;
+
+public class MainActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener, OnStartDragListener, OnKeyEvents {
+    private Toolbar toolbar;
+    private LinearLayout mToolbarContainer;
+    private String mSelectedConversion;
+    private DrawerLayout mDrawerLayout;
+    private ActionBarDrawerToggle mDrawerToggle;
+
+    private RecyclerView mRecyclerView;
+    private DrawerRecyclerViewAdapter mAdapter;
+    private RecyclerView.LayoutManager mLayoutManager;
+    private String[] mDrawerRecyclerViewDataset;
+    private ItemTouchHelper mItemTouchHelper;
+    private LinearLayout linearLayout;
+    private int DY = 5;
+    private boolean spinDirection = true;
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        EventBus.register(this);
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        EventBus.unregister(this);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mToolbarContainer = (LinearLayout) findViewById(R.id.toolbarContainer);
+        toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        setSpinner();
+        setNavigationDrawer();
+    }
+
+    private void setRecyclerView() {
+        mRecyclerView = (RecyclerView) findViewById(R.id.drawer_recycler_view);
+        mRecyclerView.setHasFixedSize(true);
+        mLayoutManager = new LinearLayoutManager(this);
+        mRecyclerView.setLayoutManager(mLayoutManager);
+        mDrawerRecyclerViewDataset = getDrawerRecyclerViewDataset();
+        mAdapter = new DrawerRecyclerViewAdapter(mDrawerRecyclerViewDataset, this, mSelectedConversion);
+        mRecyclerView.setAdapter(mAdapter);
+
+        ItemTouchHelper.Callback callback = new SimpleItemTouchHelperCallback(mAdapter);
+        mItemTouchHelper = new ItemTouchHelper(callback);
+        mItemTouchHelper.attachToRecyclerView(mRecyclerView);
+
+        linearLayout = (LinearLayout) findViewById(R.id.inflated_content_main);
+    }
+
+    private void setNavigationDrawer() {
+        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
+        mDrawerToggle = setupDrawerToggle();
+        mDrawerLayout.addDrawerListener(mDrawerToggle);
+    }
+
+    private String[] getDrawerRecyclerViewDataset() {
+        String[] drawerRecyclerViewDataset;
+        SharedPreferences sharedPref = getBaseContext().getSharedPreferences("Unito", Context.MODE_PRIVATE);
```

Comment:
@grishmashrestha move the hard coded strings either to strings.xml. Better create a class and define all the constants there.